operator/operator_server.go

```diff
@@ -0,0 +1,48 @@
+package operator
+
+// since operator is a serverless component on its own, its associated server
+// is only used for hosting the /metrics, /livez and /readyz endpoints
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/grafana/grafana-app-sdk/health"
+)
+
+type OperatorServer struct {
+	Mux         *http.ServeMux
+	HealthCheck health.HealthCheck
+	Port        int
+}
+
+func (s *OperatorServer) RegisterMetricsHandler(handler http.Handler) {
+	s.Mux.Handle("/metrics", handler)
+}
+
+func (s *OperatorServer) registerHealthHandlers() error {
+	s.Mux.Handle("/livez", http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		_, _ = w.Write([]byte("ok"))
+		return
+	}))
+	s.Mux.Handle("/readyz", s.HealthCheck.HTTPHandler())
```

Comment:
NIT – since the server already knows about the existence of healthchecks, we don't need to delegate HTTP handler implementation to it, we can just define it here, same as liveness handler:
```suggestion
	s.Mux.Handle("/readyz", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		if err := s.HealthCheck.Status(r.Context()); err != nil {
				_, _ = w.WriteHeader(http.StatusInternalServerError)
				// TODO: we can add the ability for HC to return detailed status, in which case we'd want to include it in the body.
				return
		}

		_, _ = w.Write([]byte("ok"))
		return
	}))
```

Reply:
That sounds better for sure, I will need to adjust the interface a little to include `Status` first.

Reply:
Oh yeah, that was a typo, sorry 😅